### DISCUSSION TITLE
Caching per request with StrawberryShake

### DISCUSSION THREAD

**Discussion by dev-in-disguise** [ChilliCream/hotchocolate#5032] [Q&A]
Hi,
when using StrawberryShake there is the possibility to enable caching via the `noStore` setting in the `.graphqlrc.json` file. 

I'm wondering if there is somewhere a setting I haven't found yet to enable caching on a per query basis. And if there isn't: is something like this planned?

**Comment by michaelstaib** [comment]
Can you elaborate on your use case?

**Reply by dev-in-disguise** [reply]
Yeah sure! 
Lets say I have one entity called `Client` which looks something like this: 

```
type Client {
  id: UUID!
  number: String!
  productType: ProductType!
  name: String!
  configuration: Configuration!
}
```

Out of this entity all properties will never change except the configuration property which has itself several properties that can be configured via some backoffice tool during runtime. 

On the client code we now have 2 queries defined for Strawberry Shake. 

```
query GetClientInfo($clientId: UUID!) {
  client(clientId: $clientId)
  {
    number,
    productType,
    name
  }
}
```

```
query GetClientConfiguration($clientId: UUID!) {
  client(clientId: $clientId)
  {
    configuration {
      depositLimit
    }
  }
}
```

In my code both queries can now be called multiple times and depending on the `noStore` configuration I have different issues.

* In case `noStore` is set to `true` then I have a network roundtrip all the time for both queries although it would suffice to only call the GraphQL API for `GetClientInfo` just once and then retrieve the value from the store. 
* In case `noStore` is set to `false` then I have the possibility that I'm working on cached data from the store which is potentially already outdated for the `GetClientConfiguration` query and thus would lead to some business errors. 


In this scenario I could of course set `noStore` to true and handle caching for the `GetClientInfo` query myself. What I'm wondering now is if StrawberryShake has the possibility to cache only the data of `GetClientInfo` but not for `GetClientConfiguration`. 

And now that I'm writing all that another alternative comes to mind: I could create a second GraphQL Client with the same configuration and the only difference being `noStore`. Personally I don't like that idea since the `schema.graphql`, `schema.extensions.graphql` and `graphqlrc.json` files would be 99% duplicates but still that could remove the necessity of writing an own cache.

Is there any recommendation how I should solve this kind of scenario?
